File: bogo/bogo.lua
function bogosort ()
    math.randomseed(os.time())
    io.write("Welcome to BogoSort in Lua!")
    io.write("Please enter the number of integers you would like to sort:\n")
    size = io.read()
    list = {}
    -- Lua issue "workaround"
    math.random(100)
    for i=0, size do
        list[i] = math.random(10)
    end 
    -- if type (list) ~= 'table' then return list end
 
    -- Fisher-Yates Knuth shuffle
    local function shuffle ()
        local rand = math.random(1,#list)
        for i=1,#list do
            list[i],list[rand] = list[rand],list[i]
            rand = math.random(1,#list)
        end
    end
 
    -- Returns true only if list is now sorted
    local function in_order ()
        local last = list[1]
        for i,v in next,list do
            if v < last then return false end
            last = v
        end
        return true
    end
 
    while not in_order() do shuffle() end
 
    return list
end

File: bogo/bogo.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <time.h>

 // GLOBALS HERE
unsigned int attempt;

// GENERATE RANDOM LIST OF NUMBERS TO SORT
void generate_list(int * a, int n) {
  int i;
  for (i = 0; i < n; i++) {
    a[i] = rand() % 10;
  }
}

// PRINT CURRENT LIST OF INTEGERS
void printList(int * a, int n) {
  int i;
  for (i = 0; i < n; i++) {
    printf("%d ", a[i]);
  }
}

// DETERMINE IF INTEGERS ARE SORTED
bool is_sorted(int * a, int n) {
  int i = n;
  while (--n >= 1) {
    if (a[n] < a[n - 1]) {
      ++attempt;
      printf("Attempt # %d: ", attempt);
      printList(a, i);
      return false;
    }
  }
  printf("CONGRATULATIONS YOU HAVE BOGO SORTED!!!");
  printf("FINAL SORTED VALUES ARE...\n");
  printList(a, i);
  return true;
}

// PREMIUM QUALITY BOGO HERE
void shuffle(int * a, int n) {
  int i, t, r;
  for (i = 0; i < n; i++) {
    t = a[i];
    r = rand() % n;
    a[i] = a[r];
    a[r] = t;
  }
}

// A CONVENIENTLY TIDY ABSTRATION
void bogo_sort(int * a, int n) {
  while (!is_sorted(a, n)) shuffle(a, n);
}

// THE MAIN SHOW
int main() {
  // SEED RAND HERE
  srand(time(0));
  
  attempt=0;
  int n;

  printf("Welcome to BOGOSORT in C! :) \n");
  printf("HOW MANY ITEMS DO YOU WANT TO SORT?\n");
  printf("ITEMS: ");
  scanf("%d", & n);

  int a[n];
  generate_list(a, n);
  bogo_sort(a, n);
  printf("\n");
  return 0;
}
